pkg/prometheus/promcfg.go

```diff
@@ -1021,6 +1027,11 @@ func (cg *ConfigGenerator) generatePodMonitorConfig(
 		})
 	}
 
+	// Add scrape class relabelings if there is any.
+	if scrapeClass != nil && len(scrapeClass.Relabelings) > 0 {
```

Comment:
(nit) same here, we should be able to call generateRelabelConfig() with nil slice.
```suggestion
	if scrapeClass != nil {
```